lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/UpdateTaskHost.cs

```diff
@@ -43,23 +42,28 @@ public sealed class UpdateTaskHost<TMapInput, TMapOutput, TResult> : ITask
         private readonly IBroadcastSender<MapInputWithControlMessage<TMapInput>> _dataAndControlMessageSender;
         private readonly IUpdateFunction<TMapInput, TMapOutput, TResult> _updateTask;
         private readonly bool _invokeGC;
+        private readonly IObserver<TResult> _resultHandler;
```

Comment:
Should you have a another interface for the ResultHandler? IObserver<TResult> is so generic. By using it, you pretty much limit your code not to use this interface for anything else. You can define your interface that inherit from IObserver<TResult>. Have another interface also makes you able to set default implementation instead of putting default logic in the driver when creating Task configuration.